src/test/java/io/cryostat/rules/RuleProcessorTest.java

```diff
@@ -122,25 +125,28 @@ void testDisableRemovesProcessorAsDiscoveryListener() {
 
     @Test
     void testSuccessfulRuleActivationWithCredentials() throws Exception {
-        HttpRequest<Buffer> request = Mockito.mock(HttpRequest.class);
-        HttpResponse<Buffer> response = Mockito.mock(HttpResponse.class);
-        Mockito.when(response.statusCode()).thenReturn(200);
-
-        Mockito.when(webClient.post(Mockito.any())).thenReturn(request);
-        Mockito.when(request.putHeaders(Mockito.any())).thenReturn(request);
-        Mockito.doAnswer(
-                        new Answer<Void>() {
-                            @Override
-                            public Void answer(InvocationOnMock invocation) throws Throwable {
-                                AsyncResult res = Mockito.mock(AsyncResult.class);
-                                Mockito.when(res.failed()).thenReturn(false);
-                                Mockito.when(res.result()).thenReturn(response);
-                                ((Handler<AsyncResult>) invocation.getArgument(1)).handle(res);
-                                return null;
-                            }
-                        })
-                .when(request)
-                .sendMultipartForm(Mockito.any(), Mockito.any());
+        RecordingOptionsBuilder recordingOptionsBuilder =
+                Mockito.mock(RecordingOptionsBuilder.class);
```

Comment:
Quick question, why are `recordingOptionsBuilder` and `recordingOptions` created with `Mockito.mock` instead of `@Mock`?

Reply:
`@Mock` applies to fields, whereas here `Mockito.mock` is creating a mock that is scoped within the test method. The variable could be pulled up to the field level, but that kind of pollutes the scope if the field isn't actually used in all/most methods.